### PR TITLE
Add LogUp-GKR benchmark

### PR DESCRIPTION
**Pull request by Al-Kindi-0**
Adds a simple end-to-end LogUp-GKR benchmark. Simplifies some of the benchmarks in the `sum-check` crate.

### CODE REVIEW

File: prover/benches/logup_gkr.rs [hunk 1]
```diff
@@ -0,0 +1,351 @@
...
+}
+
+criterion_group!(lagrange_kernel_group, prove_with_lagrange_kernel);
+criterion_main!(lagrange_kernel_group);
+
+// LogUpGkrSimple
+// =================================================================================================
+
+#[derive(Clone, Debug)]
+struct LogUpGkrSimple {
```
Comment (irakliyk) [inline prover/benches/logup_gkr.rs:52]: nit: seems like this is a mock execution trace, right? If so, maybe it should be called `LogUpGkrSimpleTrace`?

File: prover/benches/logup_gkr.rs [hunk 2]
```diff
@@ -0,0 +1,351 @@
...
+
+        for i in 0..4 {
+            values_1[i + 4] = BaseElement::ONE;
+        }
+
+        let mut values_2: Vec<BaseElement> = (0..trace_len).map(|_idx| BaseElement::ZERO).collect();
+
+        for i in 0..4 {
+            values_2[i + 4] = BaseElement::ONE;
+        }
```
Comment (irakliyk) [inline prover/benches/logup_gkr.rs:85]: Could we add a short explanation of what is happening below? (i.e., what do the 5 columns in the table are meant to contain).

File: prover/benches/logup_gkr.rs [hunk 3]
```diff
@@ -0,0 +1,351 @@
...
+    crypto::{hashers::Blake3_256, DefaultRandomCoin},
+    math::{fields::f64::BaseElement, ExtensionOf, FieldElement},
+    matrix::ColMatrix,
+    DefaultConstraintEvaluator, DefaultTraceLde, Prover, StarkDomain, Trace, TracePolyTable,
+};
+
+const TRACE_LENS: [usize; 2] = [2_usize.pow(20), 2_usize.pow(21)];
+const AUX_TRACE_WIDTH: usize = 2;
+
+fn prove_with_lagrange_kernel(c: &mut Criterion) {
```
Comment (irakliyk) [inline prover/benches/logup_gkr.rs:26]: Could we add comments to this function to describe what it is primarily benchmarking? I believe it is the full end-to-end proving process for a trace table which has 5 columns and some very simple constraints. But it does use LogUp-GKR (though with a relatively simple input layer).

**Review by irakliyk — APPROVED** [review]
Looks good! Thank you! I left a few doc-related comments inline.

File: prover/benches/logup_gkr.rs (hunk 1 above)
Comment (Al-Kindi-0) [reply prover/benches/logup_gkr.rs]: Changed

File: prover/benches/logup_gkr.rs (hunk 2 above)
Comment (Al-Kindi-0) [reply prover/benches/logup_gkr.rs]: Added

File: prover/benches/logup_gkr.rs (hunk 3 above)
Comment (Al-Kindi-0) [reply prover/benches/logup_gkr.rs:36]: Added

**Review by irakliyk — APPROVED** [review]
All looks good! Thank you!

**Pull request merged**